CleanWeatherTests/Scenes/FavouriteCities/FavouriteCitiesInteractorTests.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  FavouriteCitiesInteractorTests.swift
+//  CleanWeatherTests
+//
+//  Created by Marek Skrzelowski on 10/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Quick
+import Nimble
+
+@testable import CleanWeather
+
+final class FavouriteCitiesInteractorTests: QuickSpec {
+
+    override func spec() {
+
+        var presenter: FakeFavouriteCitiesPresenter!
+        var worker: FakeFavouriteCitiesWorker!
+        var router: FakeFavouriteCitiesRouter!
+        var interactor: FavouriteCitiesInteractorImpl!
+
+        beforeEach {
+            presenter = FakeFavouriteCitiesPresenter()
+            worker = FakeFavouriteCitiesWorker()
+            router = FakeFavouriteCitiesRouter()
+            interactor = FavouriteCitiesInteractorImpl(presenter: presenter, worker: worker, router: router)
+        }
+
+        describe("geting cities") {
+
+            beforeEach {
+                interactor.getCities()
+            }
+
+            it("should call presenter to show spinner") {
+                expect(presenter.toggleSpinnerStateCalled).to(beTrue())
+            }
+
+            it("should call worker to fetch all cities") {
+                expect(worker.fetchAllCitiesCalled).to(beTrue())
+            }
+
+            context("on success response") {
+
+                beforeEach {
+                    worker.fetchAllCitiesCompletion?(.success(Mock.allCities))
+                    worker.fetchFavouriteCitiesCompletion?(.success(Mock.favouriteCities))
+                }
+
+                it("should call presenter to hide spinner") {
+                    expect(presenter.toggleSpinnerStateCalled).to(beFalse())
+                }
+
+                it("should call presenter to present all cities") {
```

Comment:
you test that presenter should NOT present cities, as favourites are in progress